src/dataIntegration/transcriptomics/createTissueSpecificModel.m

```diff
@@ -1,701 +1,292 @@
-Transcripts.Locus = [];
-Transcripts.Data = {};
-for i = 1:length(Results.AltSplice)
-    num = find(Locus==Results.AltSplice(i));
-    transcripts = unique(ExpressionData.Transcript(num));
-    for j = 1:length(transcripts)
-        Transcripts.Locus = [Transcripts.Locus;Results.AltSplice(i)];
-    end
-    Transcripts.Data = [Transcripts.Data;transcripts];
-end
-
-% Determining each transcripts expression using a similar threshold as
-% before
-for i = 1:length(Transcripts.Data)
-    loc = strmatch(Transcripts.Data{i},ExpressionData.Transcript,'exact');
-    for j = 1:length(loc)
-        total(j) = sum(ExpressionData.Data(loc(j),:));
-        ExpressionData.Locus(loc(j)) = 0;
-    end
-    cap = length(loc)*11*2;
-    if sum(total)/cap >= ExpThreshold
-        Transcripts.Expression(i,1) = 1;
-    else
-        Transcripts.Expression(i,1) = 0;
-    end
-end
-
-% If expression of transcripts of one locus are the same, added as if no
-% alternative splicing occurs for simplicity
-Transcripts.Expressed = [];
-Transcripts.UnExpressed = [];
-mem_tran = Transcripts.Locus;
-for i = 1:length(Transcripts.Locus)
-    if Transcripts.Locus(i) > 0
-        locus = Transcripts.Locus(i);
-        loc = find(Transcripts.Locus==locus);
-        sumExp = 0;
-        for j = 1:length(loc)
-            sumExp = sumExp+Transcripts.Expression(loc(j));
-            Transcripts.Locus(loc(j)) = 0;
-        end
-        if sumExp == 0
-            Transcripts.UnExpressed = [Transcripts.UnExpressed;locus];
-        elseif sumExp == length(loc)
-            Transcripts.Expressed = [Transcripts.Expressed;locus];
-        end
-    end
-end
-Transcripts.Locus = mem_tran;
+function [tissueModel,Rxns] = createTissueSpecificModel(model, expressionData,exRxnRemove,solver,funcModel,options,param)
+                                                  
+% Creates draft tissue specific model from mRNA expression data
+%
+% USAGE:
+%
+%    [tissueModel, Rxns] = createTissueSpecificModel(model, expressionData, exRxnRemove, solver, funcModel, options,param)
+%
+%INPUTS
+%	model                   model strusture
+%	expressionData          mRNA expression data structure
+%       .gene               	cell array containing GeneIDs in the same
+%                               format as model.genes
+%       .value                  Vector containing corresponding expression
+%                               value (FPKM)
+%
+%OPTIONAL INPUTS
+%	exRxnRemove             Names of exchange reactions to remove
+%                           (Default = [])
+%	solver                  Use either 'GIMME','iMAT','INIT','GIMME','mCADRE','fastCore'
+%                           (Default = 'GIMME')
+%	funcModel               1 - Build a functional model having only reactions
+%                           that can carry a flux (using a consistency check), 0 - skip this
+%                           step (Default = 0)
+%	options                 1- user defines the parameters for running the extraction
```

Comment:
I think options can be omitted. if params is provided, the defaults should be overwritten. This is kind of a replication You have to indicate that you want the defaults to be overwritten AND provide the overriding values. In myy opinion just providing the values should be sufficient.